core/src/style/textStyle.cpp

```diff
@@ -111,6 +111,18 @@ auto TextStyle::applyRule(const DrawRule& _rule, const Properties& _props) const
         }
     }
 
+    if (_rule.get(StyleParamKey::repeat_group, p.labelOptions.repeatGroup)) {
+        StyleParam::Width repeatDistance;
+        if (_rule.get(StyleParamKey::repeat_distance, repeatDistance)) {
+            p.labelOptions.repeatDistance = repeatDistance.value * m_pixelScale;
+        }
+    } else {
+        // TODO: default to 'draw.key'
+    }
+
+    // TBD: should avoid allocation for combined string
+    p.labelOptions.repeatGroup += "/" + p.text;
+
```

Comment:
@karimnaaji yup you are right .. by what I meant :).
@hjanetzek yup that was a generic comment not on that line and also a
future potential optimization.
On Jan 18, 2016 2:32 PM, "Hannes Janetzek" <notifications@github.com> wrote:

> In core/src/style/textStyle.cpp
> <https://github.com/tangrams/tangram-es/pull/465#discussion_r50035989>:
>
> > @@ -111,6 +111,18 @@ auto TextStyle::applyRule(const DrawRule& _rule, const Properties& _props) const
> >          }
> >      }
> >
> > +    if (_rule.get(StyleParamKey::repeat_group, p.labelOptions.repeatGroup)) {
> > +        StyleParam::Width repeatDistance;
> > +        if (_rule.get(StyleParamKey::repeat_distance, repeatDistance)) {
> > +            p.labelOptions.repeatDistance = repeatDistance.value * m_pixelScale;
> > +        }
> > +    } else {
> > +        // TODO: default to 'draw.key'
> > +    }
> > +
> > +    // TBD: should avoid allocation for combined string
> > +    p.labelOptions.repeatGroup += "/" + p.text;
> > +
>
> @karimnaaji <https://github.com/karimnaaji> Is it necessary that all text
> labels have a repeatGroup? Most point features have a unique position and
> label. For these it would be good to (at least be able to) avoid the per
> frame check.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tangrams/tangram-es/pull/465/files#r50035989>.
>